firefox-ios/Client/TabManagement/TabManagerImplementation.swift

```diff
@@ -322,6 +322,16 @@ class TabManagerImplementation: LegacyTabManager, Notifiable, WindowSimpleTabsPr
         saveCurrentTabSessionData()
 
         willSelectTab(url)
+
+        let previous = previous ?? selectedTab
+
+        previous?.metadataManager?.updateTimerAndObserving(state: .tabSwitched, isPrivate: previous?.isPrivate ?? false)
+        tab.metadataManager?.updateTimerAndObserving(state: .tabSelected, isPrivate: tab.isPrivate)
+
+        _selectedIndex = tabs.firstIndex(of: tab) ?? -1
+
+        preserveTabs()
+
         Task(priority: .high) {
```

Comment:
@OrlaM Doesn't this async/Task still create the potential for a race condition? If two tabs were selected very quickly in sequence it doesn't appear that there's any guarantee that `selectTabWithSession` for the first tab would complete before the 2nd tab is selected, or that they'd necessarily even happen in order. TBH I'm not 100% clear on how all of this is actually working together however so maybe I'm missing something key. But the `selectTabWithSession` seems like it is still responsible for some fairly important logic (notifying all of the delegates etc).

Reply:
select tab with session is run on the main actor so it shouldn't ever run out of order, but just in case i have added a check to ensure that the tab is still the selected tab when it is run.

Reply:
I'm fairly certain that there isn't a guarantee that the code within `Task` (the call to `selectTabWithSession`) will execute before a subsequent call to `selectTab()`, even if the individual functions are `@MainActor`. But I could be incorrect or overlooking something. Regardless this PR seems like an improvement to how this is currently working. 👍 Thank you for also removing the recent history panel hack.